Sources/AST/Expression/ExternalCall.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ExternalCall.swift
+//  AST
+//
+//  Created by Catalin Craciun on 10/11/2018.
+//
+import Source
+import Lexer
+
+/// A call to an external function
+public struct ExternalCall: ASTNode {
+  public var configurationParameters: [FunctionArgument]
+  public var functionCall: BinaryExpression
+  public var returnsOptional: Bool // used for representing call?
+  public var forced: Bool // used for representing call!
```

Comment:
Regarding this + the previous line (`returnsOptional`)

Considering how you have two booleans but only 3 states, I think it would be better to represent this with an enum. Specifically to prevent an error where you might set `force = true` and `returnsOptional = true`.

```swift
public struct ExternalCall: ASTNode {
  // ...
  enum Mode {
    case doRequired, gracefullyOptional, force
  }
  // ...
}
```
names subject to your opinion ¯\_(ツ)_/¯

Reply:
I agree a three-state enum is better. By the way, in existing code, there is a similar situation for trait types (`struct`, `contract`). I believe the "kind" of trait is stored as the token (i.e. `.struct` or `.contract`). If we took the same approach we'd store the token `?`, `!`, or dummy for the default mode.